src/AspNetCore/Src/inprocessapplication.cxx

```diff
@@ -189,13 +293,32 @@ http_flush_response_bytes(
         pfCompletionExpected);
     return hr;
 }
-// End of export 
 
+EXTERN_C __MIDL_DECLSPEC_DLLEXPORT
+HRESULT
+http_enable_websockets(
```

Comment:
I think we don't need this function. We can always enable duplex in Forwarding_Handler::OnExecuteRequestHandler for Win8 and above

Reply:
WIth the OnAsyncCompletion, we can't have concurrent reads and writes going on at the same time right? Hence only enabling duplex mode when we enable websockets.

Reply:
On win8 and above, you can process the request and write response with this duplex mode like what you did. Only on Win7, you cannot use it.  This is the internal implementation details on how you handle the request

Reply:
I'd rather keep the non websocket scenario the same across each platform. There should be no point in enabling duplex mode if we are only doing one read and write operation at a time.